targets/targets.json

```diff
@@ -2589,6 +2589,28 @@
             "STM32G431xx"
         ]
     },
+    "NUCLEO_G431KB": {
+        "inherits": [
+            "MCU_STM32G431xB"
+        ],
+      "overrides": {
+        "clock_source": "USE_PLL_HSI",
+        "lse_available":  0
+      },
+      "extra_labels_add": [
+        "STM32G431KB"
+      ],
+      "macros_add": [
+        "STM32G431KB"
+      ],
+      "device_has_add": [
+        "USBDEVICE"
```

Comment:
Please remove it as there is no user USB connector on the NUCLEO_G431KB.
You could add it in your local mbed_app json file for your custom board

Reply:
I'd like to argue against that, since the USB connector pins are exposed on the NUCLEO board itself (PA11,PA12).
And it's simple operation to solder on (or breadboard) as USB cable towards those PINs.
As a NUCLEO board user, I would be surprised if USB Device didn't work out of the box and I had to modify targets.json to enable it. The risk is that it is seen as a bug (I.e. User solders USB cable to PA11/12, connects to PC, and creates a USBSerial tty object in main.cpp => Doesn't work unless he adds USBDEVICE to targets.json)

Reply:
I understand, but no :-)
In mbed wording, "USBDEVICE" means you have a user USB connector... not you support USB device...


@0xc0170 
Maybe this can be improved ?

Reply:
@MarceloSalazar

Reply:
Ok, removed USBDEVICE from targets.json for G431KB